toolium/utils/dataset.py

```diff
@@ -221,25 +237,51 @@ def _replace_param_date(param, language):
     """
     Transform param value in a date after applying the specified delta.
     E.g. [TODAY - 2 DAYS], [NOW - 10 MINUTES]
+    An specific format could be defined in the case of NOW this way: NOW('THEFORMAT')
+    where THEFORMAT is any valid format accepted by the python 
+    [datetime.strftime](https://docs.python.org/3/library/datetime.html#datetime.date.strftime) function 
 
     :param param: parameter value
     :param language: language to configure date format for NOW and TODAY
     :return: tuple with replaced value and boolean to know if replacement has been done
     """
-    date_format = '%d/%m/%Y %H:%M:%S' if language == 'es' else '%Y/%m/%d %H:%M:%S'
-    date_day_format = '%d/%m/%Y' if language == 'es' else '%Y/%m/%d'
-    date_matcher = re.match(r'\[(NOW|TODAY)\s*([\+|-]\s*\d+)\s*(\w+)\s*\]', param)
-    new_param = param
-    param_replaced = False
-
-    if date_matcher and len(date_matcher.groups()) == 3:
-        configuration = dict([(date_matcher.group(3).lower(), int(date_matcher.group(2).replace(' ', '')))])
-        now = (date_matcher.group(1) == 'NOW')
-        reference_date = datetime.datetime.utcnow() if now else datetime.datetime.utcnow().date()
-        replace_value = reference_date + datetime.timedelta(**configuration)
-        new_param = replace_value.strftime(date_format) if now else replace_value.strftime(date_day_format)
-        param_replaced = True
-    return new_param, param_replaced
+    def _split_param():
+        date_matcher = re.match(r'\[(NOW(?:\((?:\'.*\'|".*")\)|)|TODAY)(?:\s*([\+|-]\s*\d+)\s*(\w+)\s*)?\]', param)
+        if date_matcher:
+            return list(date_matcher.groups())
+
+    def _offset_datetime(amount, units):
+        now = datetime.datetime.utcnow()
+        if not amount:
+            return now
+        if not units:
+            return now
```

Comment:
I like to evaluate every single error separately and leaving the happy path to the end. Though verbose, it makes really easy to read and maintain the code. Besides, I'm not sure if joining checks in a single condition reduces complexity. 

Said this, because that's the style of this project code, I'll change it.